### ISSUE TITLE
== is broken for Colors

### ISSUE THREAD

**Issue by PragTob** [shoes/shoes-deprecated#175]
Run the following script in shows:

```
Shoes.app do
  c1 = rgb 255, 120, 0
  c2 = rgb 0, 255, 120

  if c1 == c2
    alert "Same!!!" 
  else
    alert "Different!!!"
  end
end
```

It will claim that the 2 colors are the same although they are not :-/ It seems that shoes only checks for the presence of all the values and not whether or not they are in the right order (e.g. assigned to red or blue). 

Unfortunately I failed to find the right place for this in the source (and fear that it is in C ;-)  ) - however if someone points me to the right place I'd try to fix it myself :-)

cheers,
Tobi

**Comment by steveklabnik** [comment]
If you run `ack -r "shoes_color_" shoes/*` you'll see all the methods relating to color.

Around [here](https://github.com/shoes/shoes/blob/develop/shoes/ruby.c#L4724) are where most of the definitions are. [Here](https://github.com/shoes/shoes/blob/develop/shoes/ruby.c#L4731) you can see

```
rb_include_module(cColor, rb_mComparable);
rb_define_method(cColor, "<=>", CASTHOOK(shoes_color_spaceship), 1);
```

`shoes_color_spaceship` is defined [here](https://github.com/shoes/shoes/blob/develop/shoes/ruby.c#L1849)

Sooooo yeah. Check that out. ;) The fix should be pretty easy.

**Comment by PragTob** [comment]
Thanks for the quick hint Steve!

I think I can fix this but we'll see - the problem is it just adds up all the values and checks whether their sum is equivalent.

In order not to break legacy code I will still add them up for determining whether one color is smaller or bigger than the other. Hope I get to it soon.

Happy new year!

**Comment by steveklabnik** [comment]
Exactly!

> In order not to break legacy code

I'm not sure anyone has ever used Color#<=>. Heh.

If you don't get to it in a week or two, I'll take a crack at it. :)

**Comment by PragTob** [comment]
This looks easy enough even for my rudimentary c-skills to fix (a question of a couple of ifs) - I bet you got more important stuff to do (writing a book etc) :-)

(And in the unlikely event of boredom you can still take a look at my hacketyhack pull requests :-D)

**Comment by steveklabnik** [comment]
Yeah, I have work to do with that, and getting Markdown into Shoes, and .... yeah. lots.

**Comment by steveklabnik** [comment]
This was fixed by #176.
